appjs/views/edit_project.js

```diff
@@ -58,6 +134,49 @@ var EditProject = BaseView.extend(require('./mixins/actions'), require('./mixins
     }, this);
   },
 
+  askToSave: function() {
+    logger.debug('got to ask to save');
+    var view = this;
+    var modalView = new BaseModalView({app: this.app, parentView: this}),
+        ret = new Promise(function(resolve, reject) {
+          modalView.once('cancel submit', function() {
+            resolve(true);
+          });
+          modalView.on('close', function() {
+            logger.debug('~~~~ got close');
+            resolve(false);
+          });
+        });
+
+    if($('#base-modal').length === 0){
+      modalView.show();
+    }
+
+    return ret;
+  },
+
+  hasUnsavedChanges: function(){
```

Comment:
This will return true if `upToDate` has not been set. You might want to change the contents of the function to `return this.upToDate === false;`